core/src/main/java/org/infinispan/protostream/WrappedMessage.java

```diff
@@ -296,15 +297,15 @@ private static void writeMessage(ImmutableSerializationContext ctx, TagWriter ou
             if (t.getClass().isEnum()) {
                ((EnumMarshallerDelegate) marshallerDelegate).encode(WRAPPED_ENUM, (Enum) t, out);
             } else {
-               ByteArrayOutputStreamEx buffer = new ByteArrayOutputStreamEx();
-               TagWriterImpl nestedCtx = TagWriterImpl.newInstanceNoBuffer(ctx, buffer);
-               marshallerDelegate.marshall(nestedCtx, null, t);
-               nestedCtx.flush();
-               out.writeBytes(WRAPPED_MESSAGE, buffer.getByteBuffer());
+               ProtobufTagMarshaller.WriteContext nestedWriter = out.subWriter(WRAPPED_MESSAGE, false);
+               marshallerDelegate.marshall(nestedWriter, null, t);
+               if (nestedWriter instanceof Closeable) {
+                  ((Closeable) nestedWriter).close();
+               }
```

Comment:
forgot to ask, don't you have to close every time? 
```suggestion
               nestedWriter.getWriter().close();
```

Reply:
Technically it is always true afaik, but this is cleaner.